src/packages/database/database_manager.py

```diff
@@ -53,6 +53,59 @@ class property, and the DatabaseManage.query() method used to perform a query.
 
     """
 
+    @classmethod
+    @CONFIG_MARKER
+    def rehash_handler(cls, data: Dict):
+        """
+        Apply new configuration data
+
+        Args:
+            data (typing.Dict): new configuration data to apply.
+
+        """
+        cls._config = data
+
+    _config: ClassVar[Dict] = {}
+
+    @classmethod
+    @CONFIG_MARKER
+    def validate_config(cls, data: Dict):
+        """
+        Validate database portion of the configuration file
+
+        Args:
+            data(typing.Dict): configuration object
+        """
+        try:
+            module_config = data['database']
+        except (KeyError, AttributeError) as ex:
+            raise ValueError("No Database configuration in file.") from ex
+
+        # Require all values to be set
+        for setting in module_config.items():
+            if setting is None:
+                raise ValueError(f"{setting} is required for instantiation")
+
+        # Host
+        if not isinstance(module_config['host'], str):
+            raise ValueError("host must be a string.")
+
+        # Port
+        if not isinstance(module_config['port'], int):
+            raise ValueError("port must be an integer.")
```

Comment:
[Question] Might it be better to keep a running list of errors and return them at the end? The fast-fail solution we have now means that users are only made aware of one config issue at a time.

Reply:
As per [pluggy's documentation](https://pluggy.readthedocs.io/en/latest/#exception-handling), 
> If any hookimpl errors with an exception no further callbacks are invoked and the exception is packaged up and delivered to any Wrappers before being re-raised at the hook invocation point:

So holding for several errors in validation is not possible.

Or are you referring to within each module have it gather exceptions and raise one at the end?